src/WebAPIContentService/Application/Controllers/MaterialsController.cs

```diff
@@ -1,38 +1,105 @@
 using Microsoft.AspNetCore.Mvc;
-using Microsoft.EntityFrameworkCore;
 using WebAPIContentService.Domain.Entities;
-using WebAPIContentService.Infra.Data.Context;
+using WebAPIContentService.Service.Interfaces;
 
 namespace WebAPIContentService.Application.Controllers
 {
     [Route("[controller]")]
     [ApiController]
     public class MaterialsController : ControllerBase
     {
-        private readonly ContentContext _context;
+        private readonly IMaterialService _materialService;
 
-        public MaterialsController(ContentContext context)
+        public MaterialsController(IMaterialService materialService)
         {
-            _context = context;
+            _materialService = materialService;
         }
 
-        // GET: /materials
         [HttpGet]
         public async Task<ActionResult<IEnumerable<Material>>> GetMaterials()
         {
-            return await _context.Materials.ToListAsync();
+            return Ok(await _materialService.GetAllMaterialsAsync());
         }
 
-        // POST: /materials
-        [HttpPost]
-        public async Task<ActionResult<Material>> PostMaterial(Material material)
+        // Consultar um material pelo Id
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Material>> GetMaterialById(int id)
+        {
+            var material = await _materialService.GetMaterialByIdAsync(id);
+
+            if (material == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(material);
+        }
+
+        [HttpGet("diretoria/{id}")]
+        public async Task<ActionResult<IEnumerable<Material>>> GetMaterialsByIdDiretoriaA(int id)
         {
-            material.CreatedAt = DateTime.Now;
-            material.UpdateAt = DateTime.Now;
-            _context.Materials.Add(material);
-            await _context.SaveChangesAsync();
+            var materials = await _materialService.GetMaterialsByIdDiretoriaAsync(id);
+
+            if (materials == null || !materials.Any())
+            {
+                return NotFound();
+            }
+
+            return Ok(materials);
+        }
+
+        [HttpGet("search")]
+        public async Task<ActionResult<IEnumerable<Material>>> GetMaterialByTitulo([FromQuery] string titulo)
+        {
+            var materials = await _materialService.GetMaterialByTituloAsync(titulo);
+
+            if (materials == null || !materials.Any())
+            {
+                return NotFound();
```

Comment:
Nesse caso, eu acho mais valido retornar a lista vazia do que um NotFound, é um conceito interessante que parece detalhe, mas faz sentido, vou mandar a explicação aqui, mas para os casos em que faz uma busca específica, tipo um ID x, vale a pena usar NotFound. Mas para caso que busca uma lista, que pode ter ou não ter, e não buscando um item específico não vale a pena usar NotFound, mas não tenho certeza sobre isso, é o que eu aprendi

**Retornar 200 OK com uma lista vazia:**

Essa abordagem indica que a consulta foi bem-sucedida, mas não há resultados correspondentes à pesquisa. Isso é apropriado quando a operação foi executada sem erros, mas simplesmente não há dados a serem retornados.
É uma boa opção quando uma lista vazia é um resultado válido e esperado. Isso evita que os consumidores da API interpretem uma resposta de erro quando não é necessária.

**Retornar 404 Not Found:**

Essa abordagem indica que o recurso procurado não foi encontrado. Geralmente, é usada quando a consulta se refere a um recurso específico e a pesquisa não encontra esse recurso.
Pode ser apropriada quando a consulta se refere a algo muito específico, como um item individual, e a ausência desse item é considerada uma situação de erro.

Reply:
Faz sentido sim, vou fazer essa mudança em todos os lugares que usa o NotFound() dessa maneira.

Reply:
Excelente pontuação!

Mantenhamos o 404 para erros de busca de conteúdo específico, ou para ausência de permissão para acessar um conteúdo, bem como o GitHub faz quando pesquisamos um repo privado sem permissão.